server/src/main/java/org/opensearch/search/SearchService.java

```diff
@@ -234,6 +235,13 @@ public class SearchService extends AbstractLifecycleComponent implements IndexEv
         Property.NodeScope
     );
 
+    public static final Setting<Boolean> SEARCH_SEGMENTS_REVERSE_ORDER_OPTIMIZATION = Setting.boolSetting(
```

Comment:
Can this be more helpful as an index setting instead? A user can choose to apply this for the indices they deem useful within a cluster.